optimade/server/entry_collections/elasticsearch.py

```diff
@@ -0,0 +1,211 @@
+import os
+
+from pathlib import Path
+
+from typing import Tuple, List, Optional, Dict, Any, Iterable, Union
+from fastapi import HTTPException
+from elasticsearch_dsl import Search
+from elasticsearch.helpers import bulk
+import json
+import os.path
+
+from optimade.filterparser import LarkParser
+from optimade.filtertransformers.elasticsearch import ElasticTransformer, Quantity
+from optimade.server.config import CONFIG
+from optimade.server.logger import LOGGER
+from optimade.models import EntryResource
+from optimade.server.mappers import BaseResourceMapper
+from .entry_collections import EntryCollection
+
+try:
+    CI_FORCE_ELASTIC = bool(int(os.environ.get("OPTIMADE_CI_FORCE_ELASTIC", 0)))
+except (TypeError, ValueError):  # pragma: no cover
+    CI_FORCE_ELASTIC = False
+
+if CONFIG.database_backend.value == "elastic" or CI_FORCE_ELASTIC:
+    from elasticsearch import Elasticsearch
+
+    CLIENT = Elasticsearch(hosts=CONFIG.elastic_hosts)
+    print("Using: Real Elastic (elasticsearch)")
+
+
+with open(Path(__file__).parent.joinpath("elastic_indexes.json")) as f:
+    INDEX_DEFINITIONS = json.load(f)
```

Comment:
I've made it a property of ElasticCollection for now; I don't know how customisable this needs to be (for e.g. provider fields) so this should be made configurable in the config overhaul PR (added note above).

Reply:
Yeah. It comes down to whether the Collection class should handle data loading as well or not, I guess?

Reply:
I guess this should actually be covered by the `use_production_backend` config option: if `True`, don't make any new indexes, otherwise make the default one based on OPTIMADE fields only (which can eventually be extended to handle provider fields). I'll do that quickly now.